src/dmd/root/filename.d

```diff
@@ -889,4 +893,77 @@ version(Windows)
 
         return F(ret);
     }
+
+}
+
+version(Posix)
+{
+
+    /**************************
+     * Takes a callable F and applies it to the result of converting
+     * `fileName` to an absolute file path (char*)
+     */
+    auto absPathThen(alias F)(const(char)* fileName)
```

Comment:
Can a `const(char)[]` parameter be used instead.

Reply:
That would complicate the code given that the Win32 API only deals in null-terminated strings.

Reply:
We would prefer all new functions have full DDoc headers including `Params` and `Returns`.  Also see https://dlang.org/dstyle.html#phobos_documentation

Reply:
Although a `const(char)[]` would complicate the code, that complication would be encapsulated in this function, and that's good IMO.

Reply:
`Loc.filename` isn't `char[]`, it's `const(char)*`. That would require a _lot_ of conversions. `const(char)*` is used all over the codebase from the time it was written in C++. Converting `const(char)*` to `char[]` back to a C string for _both_ the Windows and Posix APIs doesn't make much sense to me.

I'm trying to fix a bug, not refactor dmd. Yes, one should refactor as one goes, but C strings are so pervasive right now I'd need dozens of PRs just to get to the point that I can fix this regression, which _seriously_ affects my workflow and is stopping us from upgrading the compiler version at work.